h/services/bulk_executor/_actions.py

```diff
@@ -171,14 +171,13 @@ def execute(self, batch, on_duplicate="continue", **_):
 
         stmt = insert(GroupMembership).values(values)
 
-        if on_duplicate == "continue":
-            # This update doesn't change the row, but it does count as it being
-            # 'updated' which means we can get the values in the "RETURNING"
-            # clause and do the select in one go
-            stmt = stmt.on_conflict_do_update(
-                index_elements=["user_id", "group_id"],
-                set_={"user_id": stmt.excluded.user_id},
-            )
+        # This update doesn't change the row, but it does count as it being
+        # 'updated' which means we can get the values in the "RETURNING"
+        # clause and do the select in one go
+        stmt = stmt.on_conflict_do_update(
+            index_elements=["user_id", "group_id"],
+            set_={"user_id": stmt.excluded.user_id},
+        )
```

Comment:
This "if" isn't reachable due to a check above. It's what we _would_ do if that check was removed, but this removes an untested (and unreachable) code path.

Reply:
Is this somehow related to the LMS authority change?

Reply:
No, but I was getting coverage warnings, and didn't want to assume I'd actually tested this